src/Microsoft.Azure.SignalR.Management/Obsolete/ServiceManager.cs

```diff
@@ -16,22 +19,39 @@ namespace Microsoft.Azure.SignalR.Management
     internal class ServiceManager : IServiceManager
     {
         private readonly RestClientFactory _restClientFactory;
-        private readonly ServiceHubContextFactory _serviceHubContextFactory;
         private readonly IServiceProvider _serviceProvider;
+        private readonly IServiceCollection _services;
         private readonly ServiceEndpoint _endpoint;
         private readonly IServiceEndpointProvider _endpointProvider;
 
-        public ServiceManager(RestClientFactory restClientFactory, ServiceHubContextFactory serviceHubContextFactory, IServiceEndpointManager endpointManager, IServiceProvider serviceProvider)
+        public ServiceManager(IServiceCollection serviceDescriptors)
         {
-            _restClientFactory = restClientFactory;
-            _serviceHubContextFactory = serviceHubContextFactory;
-            _serviceProvider = serviceProvider;
+            _services = serviceDescriptors;
+            _serviceProvider = _services.BuildServiceProvider(); ;
+            _restClientFactory = _serviceProvider.GetService<RestClientFactory>();
```

Comment:
is it possible to be null? how about GetRequiredService`
Same to next line

Reply:
Just think that if null, then error will be thrown when we need them. We don't always need `RestClientFactory` during some test scenarios.